### PR TITLE
rust: avoid clobbering LIBS

### PR DESCRIPTION
**Pull request by dkm**
Save LIBS around calls to AC_SEARCH_LIBS to avoid clobbering $LIBS.

### CODE REVIEW

**Review by tschwinge — APPROVED** [review]
Oh, are the `AC_SEARCH_LIBS`s currently affecting the general `LIBS` variable?  In that case, your proposed changes seems correct to me (which files regenerated).


> Also, do you know why this is in the top-level .ac and not in the gcc/ one?

So that we can still silently/verbosely "Disable Rust if we are missing some required C libraries for the Rust runtime", see further down in `configure.ac`, where `missing_rust_dynlibs` is used.

**Comment by tschwinge** [comment]
Indeed with your change, on a Ubuntu 20.04 system, the top-level `config.status` changes as follows:

```
@@ -951 +951 @@
-S["CRAB1_LIBS"]=" -ldl -lpthread"
+S["CRAB1_LIBS"]="-lpthread -ldl "
@@ -1021 +1021 @@
-S["LIBS"]="-lpthread -ldl "
+S["LIBS"]=""
```

..., which seems correct to me.

```
Reviewed-by: Thomas Schwinge <tschwinge@baylibre.com>
Tested-by: Thomas Schwinge <tschwinge@baylibre.com>
```

**Comment by dkm** [comment]
Thanks. This was directed by discussion on IRC. This is a common pattern in gcc/configure.ac.

> So that we can still silently/verbosely "Disable Rust if we are missing some required C libraries for the Rust runtime", see further down in configure.ac, where missing_rust_dynlibs is used.

oh, ok, thanks. I'll forward this reply on IRC where the question was asked. I'll send the patch on gcc-patches@ later today

**Comment by dkm** [comment]
I've cleaned the log, generated the `configure` file, added a small comment so that we don't have again the question "why is it in the toplevel makefile?" and I've added the `Reviewed-by/Tested-by` footer. I'll wait for your review before merging it and sending the same change to gcc-patches@.

**Review by tschwinge — APPROVED** [review]
Looks good, just one boring item: please re-regenerate `configure`, so that "added a small comment" also gets propagated there.

**Review by tschwinge — APPROVED** [review]

**Comment by dkm** [comment]
Looks like I should also update the commit checker, it doesn't like `Tested-By` footer.

**Comment by dkm** [comment]
This has been merged upstream as r15-3443-gda3a2985fff39e

**Comment by CohenArthur** [comment]
well done :) thanks

**Pull request merged**
